text/0000-associated-type-bounds.md

```diff
@@ -100,6 +102,53 @@ Meanwhile, the surface syntax `dyn Trait<AssociatedType: Bounds>` desugars into
 `dyn Trait<AssociatedType = T>` where `T` is a named type variable `T` with the
 bound `T: Bounds`.
 
+## The desugaring for associated types
+
+In the case of an associated type having a bound of the form:
+
+```rust
+trait TraitA {
+    type AssocA: TraitB<AssocB: TraitC>;
+}
+```
+
+we desugar to an anonymous associated type for `AssocB`, which corresponds to:
+
+```rust
+trait TraitA {
+    type AssocA: TraitB<AssocB = Self::AssocA_0>;
+    type AssocA_0: TraitC; // Associated type is Unnamed!
+}
+```
+
+## Notes on the meaning of `impl Trait<Assoc: Bound>`
+
+Note that in the context `-> impl Trait<Assoc: Bound>`, since the
+`Trait` is existentially quantified, so is in effect also the `Assoc`.
```

Comment:
This could use a bit of grammar cleanup for clarity. How about: since the `Trait` is existentially quantified, the `Assoc` is as well.